javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,103 @@
+let name = [];
+let todoList = [];
+function getReply(command) {
+  if (command.toLowerCase().includes("hello my name is")) {
+    const newName = command.split(" ")[4];
+    const commandWords = command.split(" ");
+    if (commandWords.length >= 5 && name.length > 0 && name.includes(newName)) {
+      return `Hello again, ${newName}`;
+    } else {
+      name.push(newName);
+      return `Nice to meet you ${newName}`;
+    }
+  }
+  if (command.toLowerCase().includes("what is my name")) {
+    return `Your name is ${name}`;
+  }
+  if (
+    command.toLowerCase().includes("add") &&
+    command.toLowerCase().includes("todo")
+  ) {
+    const todoId = command.indexOf("todo");
+    const todoItem = command.split(" ")[1];
+    todoList.push(todoItem);
+    return `${todoItem} added to your todo`;
+  }
+  if (
+    command.toLowerCase().includes("remove") &&
+    command.toLowerCase().includes("todo")
+  ) {
+    const todoId = command.indexOf("todo");
+    const todoItem = command.split(" ")[1];
+    todoList.splice(todoList.indexOf(todoItem), 1);
+    return `${todoItem} removed from your todo`;
+  }
+  if (command.toLowerCase().includes("what is on my todo?")) {
+    return `You have ${todoList.length} todos - ${todoList}`;
+  }
+  if (command.toLowerCase().includes("what day is it today?")) {
+    const today = new Date();
+    const date = today.getDate();
+    const month = today.getMonth() + 1;
+    const year = today.getFullYear();
+    return `${date}.${month}.${year}`;
+  }
+
+  if (command.startsWith("What is ")) {
+    if (command.includes("+")) {
+      let numbers = command.replace("What is ", "").split(" + ");
+      let sum = parseInt(numbers[0]) + parseInt(numbers[1]);
+      return sum;

```

Comment:
Nice job here. 2 tips (you don't need to change your code)

- To avoid some repeated code, you could parse the `numbers` and the `operator` once at the top, then use a `switch` for every case.
- The other one is to use regexes, which are very powerful, but pretty hard to debug.